nano/lib/async.hpp

```diff
@@ -49,12 +49,11 @@ class cancellation
 	};
 
 public:
-	void emit (asio::cancellation_type type = asio::cancellation_type::all)
+	auto emit (asio::cancellation_type type = asio::cancellation_type::all)
```

Comment:
Does the returned value need to be used anywhere?

Reply:
It's nice to have and I just added a test that uses this functionality